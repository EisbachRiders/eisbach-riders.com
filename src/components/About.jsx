import React from 'react'
import PropTypes from 'prop-types'
import Img from 'gatsby-image'
import { useTranslation } from 'react-i18next'
import { makeStyles } from '@material-ui/styles'
import Typography from '@material-ui/core/Typography'
import Hidden from '@material-ui/core/Hidden'

const useStyles = makeStyles(theme => ({
  root: {
    position: 'relative',
    paddingTop: 30,
    paddingBottom: 30,
    paddingLeft: 15,
    paddingRight: 15,
    background: theme.status.greyBlue,
    display: 'flex',
    alignItems: 'center',
    [theme.breakpoints.up('sm')]: {
      paddingLeft: 60,
      paddingRight: 60,
    },
    [theme.breakpoints.up('md')]: {
      paddingTop: 60,
      paddingBottom: 60,
      paddingLeft: 90,
      paddingRight: 90,
    },
    [theme.breakpoints.up('lg')]: {
      paddingTop: 45,
      paddingBottom: 90,
      paddingLeft: 170,
      paddingRight: 170,
    },
  },
  textContainer: {
    zIndex: 2,
    flexBasis: '100%',
    paddingBottom: 30,
    paddingRight: 15,
    paddingLeft: 15,
    [theme.breakpoints.up('sm')]: {
      flexBasis: '60%',
    },
    [theme.breakpoints.up('md')]: {
      marginBottom: 60,
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'space-between',
    },
    [theme.breakpoints.up('lg')]: {
      paddingTop: 45,
      paddingRight: 60,
      marginBottom: 90,
    },
  },
  imgContainer: {
    flexBasis: '40%',
    marginBottom: 45,
    zIndex: 2,
  },
  sideText: {
    position: 'relative',
    float: 'right',
    marginTop: -100,
    marginRight: 0,
    transform: 'rotate(-270deg)',
    WebkitTransform: 'rotate(-270deg)',
    MozTransform: 'rotate(-270deg)',
    MsTransform: 'rotate(-270deg)',
    OTransform: 'rotate(-270deg)',
    filter: 'progid:DXImageTransform.Microsoft.BasicImage(rotation=3)',
    [theme.breakpoints.up('sm')]: {
      marginTop: -150,
      marginRight: 0,
    },
    [theme.breakpoints.up('md')]: {
      marginTop: -150,
      marginRight: 0,
    },
  },
  img: {
    width: 200,
    height: 150,
    margin: '0 auto',
    [theme.breakpoints.up('md')]: {
      width: '80%',
      height: '80%',
    },
  },
  backgroundImg: {
    zIndex: 1,
    left: 0,
    bottom: 0,
    width: '100%',
    objectFit: 'fill',
    height: 150,
  },
  title: {
    textTransform: 'uppercase',
    fontWeight: 700,
    fontSize: 16,
    [theme.breakpoints.up('md')]: {
      fontSize: 36,
    },
  },
  text: {
    paddingTop: 30,
    color: theme.palette.common.black,
    fontSize: 12,
    [theme.breakpoints.up('md')]: {
      fontSize: 14,
    },
  },
}))

function About({ img, waveImg }) {
  const classes = useStyles()
  const { t } = useTranslation()
  return (
    <div className={classes.root} id="about">
      <div className={classes.textContainer}>
        <Typography className={classes.title} variant="h5">
          {t('about.sectionTitle1')}
        </Typography>
        <Typography className={classes.title} variant="h5">
          {t('about.sectionTitle2')}
        </Typography>
        <Typography className={classes.text} variant="body1">
          {t('about.message')}
        </Typography>
      </div>
      <Hidden xsDown>
        <div className={classes.imgContainer}>
          <Img
            fluid={img.node.fluid}
            alt="two surfers in ocean"
            className={classes.img}
          />
          {/* <Typography className={classes.sideText}>
            Surf Accessories from Munich
          </Typography> */}
        </div>
      </Hidden>
      <Img
        fluid={waveImg.node.fluid}
        alt="wave background"
        className={classes.backgroundImg}
        imgStyle={{ objectFit: 'fill' }}
        style={{ position: 'absolute' }}
      />
    </div>
  )
}

About.propTypes = {
  img: PropTypes.object.isRequired,
  waveImg: PropTypes.object.isRequired,
}

export default About
