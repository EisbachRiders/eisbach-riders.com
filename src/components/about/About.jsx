import React from 'react'
import PropTypes from 'prop-types'
import AboutDetail from './AboutDetail'
import AboutHero from './AboutHero'
import Founder from './Founder'

function About({ img }) {
  return (
    <div>
      <AboutHero img={img[3]} />

      <AboutDetail img={img} />

      <Founder img={[img[5], img[4]]} />
    </div>
  )
}

About.propTypes = {
  img: PropTypes.array.isRequired,
}

export default About
