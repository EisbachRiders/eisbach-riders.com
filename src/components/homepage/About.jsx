import React from 'react'
import PropTypes from 'prop-types'
import Img from 'gatsby-image'
import { useTranslation } from 'react-i18next'
import { makeStyles } from '@material-ui/styles'
import Typography from '@material-ui/core/Typography'
import Hidden from '@material-ui/core/Hidden'
import BrushStroke from '../ui/BrushStroke'
import Container from '../ui/Container'

const useStyles = makeStyles(theme => ({
  root: { display: 'flex' },
  container: {
    flexBasis: '100%',
    [theme.breakpoints.up('sm')]: {
      flexBasis: '50%',
    },
    [theme.breakpoints.up('md')]: {
      paddingRight: 75,
      paddingTop: 30,
      paddingBottom: 30,
    },
    [theme.breakpoints.up('lg')]: {
      paddingTop: 45,
    },
  },
  margin: {
    marginLeft: '15%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: 0,
    },
  },
  imgContainer: {
    flexBasis: '50%',
  },
  img: {
    height: '100%',
    [theme.breakpoints.up('md')]: {
      height: 400,
      width: '100%',
    },
  },
  text: {
    color: theme.palette.common.black,
    fontSize: 12,
    [theme.breakpoints.up('md')]: {
      fontSize: 14,
    },
  },
}))

function About({ img }) {
  const classes = useStyles()
  const { t } = useTranslation()
  return (
    <div className={classes.root}>
      <Container className={classes.container}>
        <div className={classes.margin}>
          <BrushStroke title={t('about.sectionTitle1')} />
        </div>
        <Typography className={classes.text} variant="body1">
          {t('about.message')}
        </Typography>
      </Container>
      <Hidden xsDown>
        <div className={classes.imgContainer}>
          <Img
            fluid={img.node.fluid}
            alt="two surfers in ocean"
            className={classes.img}
          />
        </div>
      </Hidden>
    </div>
  )
}

About.propTypes = {
  img: PropTypes.object.isRequired,
}

export default About
