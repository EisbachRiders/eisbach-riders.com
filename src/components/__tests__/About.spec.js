import { render } from 'react-testing-library'
import About from '../About'

describe('About', () => {
  it('renders correctly', () => {
    const mockProps = {
      img: {},
      waveImg: {},
    }
    const component = render(<About {...mockProps} />)
    expect(component).toMatchSnapshot()
  })
})
