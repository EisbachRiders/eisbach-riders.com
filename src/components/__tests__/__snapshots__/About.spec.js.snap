// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Hook-root-2v6cfj"
        id="about"
      >
        <div
          class="Hook-textContainer-17qwi1r"
        >
          <h5
            class="MuiTypography-root-1 MuiTypography-h5-17 Hook-title-u0uo6s"
          >
            about.sectionTitle1
          </h5>
          <h5
            class="MuiTypography-root-1 MuiTypography-h5-17 Hook-title-u0uo6s"
          >
            about.sectionTitle2
          </h5>
          <p
            class="MuiTypography-root-1 MuiTypography-body1-10 Hook-text-1fyv5yu"
          >
            about.message
          </p>
        </div>
        <div
          class="Hook-imgContainer-5tx6ux"
        >
          <div
            class="Hook-img-ldoh2k gatsby-image-wrapper"
            style="position: relative; overflow: hidden;"
          >
            <div
              style="width: 100%;"
            />
            <picture>
              <source />
              <img
                alt="two surfers in ocean"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 0;"
              />
            </picture>
            <noscript>
              &lt;picture&gt;&lt;img src="" alt="two surfers in ocean" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
            </noscript>
          </div>
          <p
            class="MuiTypography-root-1 MuiTypography-body1-10 Hook-sideText-esk9qf"
          >
            Surf Accessories from Munich
          </p>
        </div>
        <div
          class="Hook-backgroundImg-3em4y gatsby-image-wrapper"
          style="position: absolute; overflow: hidden;"
        >
          <div
            style="width: 100%;"
          />
          <picture>
            <source />
            <img
              alt="wave background"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 0;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;img src="" alt="wave background" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Hook-root-2v6cfj"
      id="about"
    >
      <div
        class="Hook-textContainer-17qwi1r"
      >
        <h5
          class="MuiTypography-root-1 MuiTypography-h5-17 Hook-title-u0uo6s"
        >
          about.sectionTitle1
        </h5>
        <h5
          class="MuiTypography-root-1 MuiTypography-h5-17 Hook-title-u0uo6s"
        >
          about.sectionTitle2
        </h5>
        <p
          class="MuiTypography-root-1 MuiTypography-body1-10 Hook-text-1fyv5yu"
        >
          about.message
        </p>
      </div>
      <div
        class="Hook-imgContainer-5tx6ux"
      >
        <div
          class="Hook-img-ldoh2k gatsby-image-wrapper"
          style="position: relative; overflow: hidden;"
        >
          <div
            style="width: 100%;"
          />
          <picture>
            <source />
            <img
              alt="two surfers in ocean"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 0;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;img src="" alt="two surfers in ocean" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
        <p
          class="MuiTypography-root-1 MuiTypography-body1-10 Hook-sideText-esk9qf"
        >
          Surf Accessories from Munich
        </p>
      </div>
      <div
        class="Hook-backgroundImg-3em4y gatsby-image-wrapper"
        style="position: absolute; overflow: hidden;"
      >
        <div
          style="width: 100%;"
        />
        <picture>
          <source />
          <img
            alt="wave background"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 0;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;img src="" alt="wave background" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
